nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestPageProvider.java

```diff
@@ -966,6 +967,35 @@ public void testMaxResultWindow() throws Exception {
         Assert.assertFalse(pp.isLastPageAvailable());
     }
 
+    /**
+     * Testing an ES page provider when not specifying limit. This shouldn't crash and be boxed by
+     * {@link ElasticSearchNxqlPageProvider#ES_MAX_RESULT_WINDOW_PROPERTY}.
+     *
+     * @since 10.3
+     */
+    @Test
+    public void iCanPerformUnlimitedQuery() throws Exception {
+        // create 10 docs
+        startTransaction();
+        for (int i = 0; i < 10; i++) {
+            DocumentModel doc = session.createDocumentModel("/", "testDoc" + i, "File");
+            doc.setPropertyValue("dc:title", "TestMe" + i);
+            doc = session.createDocument(doc);
+        }
+        TransactionHelper.commitOrRollbackTransaction();
+        waitForCompletion();
+        startTransaction();
+
+        PageProviderService pps = Framework.getService(PageProviderService.class);
+        PageProviderDefinition ppdef = pps.getPageProviderDefinition("NXQL_PP_UNLIMITED");
+        HashMap<String, Serializable> props = new HashMap<>();
```

Comment:
`Map<String, Serializable>`?

Reply:
Indeed, I won't change it because master code as already been merged :(. Hopefully its test.